### PR TITLE
Bulletin for Updates to Navigation API

### PR DESCRIPTION
**Pull request by jpmedley**

### CODE REVIEW

**Comment by jpmedley** [comment]
@tomayac, @jakearchibald FYI

**Comment by jakearchibald** [comment]
I'll review this. I'm also working on the changes for the article.

**Comment by jpmedley** [comment]
>I'll review this. I'm also working on the changes for the article.

FYI, we do bulletins like this now so that there's no rush on the article. There is a bit of a rush on this though. It should be up before Thursday, August 4.

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:18]:
```suggestion
Chrome 105 introduces two new methods on the `NavigateEvent` of the Navigation API (introduced in 102) to improve on methods that have proved problematic in practice. `intercept()`, which let's developers control the state following the navigation, replaces `transitionWhile()`, which proved difficult to use. The `scroll()` method, which scrolls to an anchor specified in the URL, replaces `restoreScroll()` which does not work for all types of navigation. 
```

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:35]:
```suggestion
This is functionally equivalent to the code below. It causes some portion of the navigation to run before API is aware that the developer intends to intercept the navigation.
```

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:44]:
```suggestion
One example where this can mess up an app is in scroll restoration logic where it captures scroll positions after the DOM change, rather than before. 
```

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:48]:
```suggestion
To replace `transitionWhile()`, the current spec introduces `NavigateEvent.intercept()`. The new method takes a handler in addition to the `focusReset` and `scrollRestoration` properties supported by `transitionWhile()`. The new handler always runs after the navigation commits, and things like scroll positions have been captured, avoiding the problems with `transitionWhile()`.
```

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:54]:
```suggestion
The `NavigateEvent.intercept()` has the same restrictions as `transitionWhile()`, as in it cannot be called on all navigation events. Cross-origin navigations cannot be intercepted, neither can cross-document traversals. Doing so will throw a [`DOMException`](https://developer.mozilla.org/docs/Web/API/DOMException) of the `"SecurityError"` type.
```

When I first read this, I thought it was suggesting that `intercept()` has more restrictions than `transitionWhile()`.

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:64]:
```suggestion
  event.intercept({
    async handler() {
      doSyncStuff();
      await doAsyncStuff();
    }
  });
```

I thought it'd be nice to bring back the example from earlier than didn't work in the old API.

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:91]:
```suggestion
By default, the browser will attempt to handle scrolling automatically, once the intercept hander has fulfilled. If you want to handle the scrolling yourself, set `scroll` to `"manual"`, then call `NavigateEvent.scroll()` when the browser should attempt to set the scroll position.
```

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:102]:
```suggestion
navigation.addEventListener("manual", event => {
  scroll: "manual",
  event.intercept({
    async handler() {
      doSyncStuff();
      // Handle scrolling earlier than by default:
      event.scroll();
      await doAsyncStuff();
    }
  });
});
```

Again, working on from the earlier example (also correcting the scroll value).

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:109]: > We hope to soon update our article on the Navigation API.

Working on it now 😄

**Review by jakearchibald — CHANGES_REQUESTED** [review]

File: site/en/blog/navigateevent-intercept/index.md
Comment (jpmedley) [reply site/en/blog/navigateevent-intercept/index.md:64]: Good call!

**Comment by jpmedley** [comment]
@jakearchibald PTAL

File: site/en/blog/navigateevent-intercept/index.md
Comment (jakearchibald) [inline site/en/blog/navigateevent-intercept/index.md:81]: doh, thanks for catching this. I saw that I needed to update this, but I guess I just forgot.

**Review by jakearchibald — APPROVED** [review]
LGTM

**Review by rachelandrew — APPROVED** [review]

**Comment by jpmedley** [comment]
**FYI, I'm waiting until closer to the beta release to merge this, Wednesday P.M. or Thursday a.m.**

File: site/en/blog/navigateevent-intercept/index.md
Comment (jpmedley) [reply site/en/blog/navigateevent-intercept/index.md:81]: np

**Comment by jakearchibald** [comment]
I should have a PR for the article tomorrow

**Comment by jpmedley** [comment]
Thanks!

**Comment by jakearchibald** [comment]
Ok it was 2 days: https://github.com/GoogleChrome/developer.chrome.com/pull/3390

**Pull request merged**
